fix: fixed counter exceding length of text

vidriduch opened the pull request:
- Fixed counter for skipping text exceeding total length of text
  when key was pressed on last line

vidriduch committed:
fix: fixed counter exceding length of text

* Fixed counter for skipping text exceeding total length of text
  when key was pressed on last line

mrshu commented on levels/utils.go:
```diff
@@ -78,6 +78,9 @@ func PrintText(text string, pretty_print bool) {
 	var counter = 0
 	for _, line := range lines {
 		counter += len(line) + 1
+		if counter > len(text) {
```
@Foxiy is it crazy to have tests for this?

mrshu commented:
Given that this bug was quite a big one I believe we need a way of avoiding it in the future. I do not see other ways of doing that than having a test suite, that would test this functionality (and also this fix).

I believe that this change (small in size by huge in terms of pain it could save us from) might be a good starting point. In all honesty, I do not even know how to asses (in other words test) whether this change even fixes the issue (since you marked it as a fix).

vidriduch committed:
update: replace print function for fmt print

* Replaced print() in print_line function for fmt print functions.
  Reason for this change is print() outputs text in a wierd way
  and character inputed this way weren't picked up by **example** golang
  functions as being outputed to stdout

vidriduch committed:
add: Added test suite file for testing of output

* Added new go test suite file for purpouse of testing correct output of gta
  and its print functions

* print_test.go - added function for testing last line counter overflow bug
                  where if user tried to skip pretty printing on last line of
                  level description panic would folow from index out of bounds

Foxiy commented:
@mrshu added new file that could server as test for all output and print line function testing and added test for this bug.

mrshu commented on levels/utils.go:
```diff
@@ -25,7 +25,7 @@ func print_line(text string, keypress chan []byte, echo_state bool) {
 	var b []byte = make([]byte, 1)
 	for _, char := range text {
-		print(string(char))
+		fmt.Printf("%s", string(char))
```
@Foxiy I assume this was necessary but seems like a win anyway, since we were already using `fmt` throughout the file.

mrshu commented:
@Foxiy Awesome, thanks a ton for taking the floor and putting together a test for this fix, it makes me feel way better when I click on that big green button.

Thanks!
